app/database/item_store.go

```diff
@@ -79,26 +79,21 @@ func (s *ItemStore) Insert(data *Item) error {
 
 // HasManagerAccess returns whether the user has manager access to all the given item_id's
 // It is assumed that the `OwnerAccess` implies manager access
-func (s *ItemStore) HasManagerAccess(user *User, itemID int64) (found, allowed bool, err error) {
+func (s *ItemStore) HasManagerAccess(user *User, itemIDs ...int64) (found bool, err error) {
+	var count int64
 
-	var dbRes []struct {
-		ItemID        int64 `sql:"column:idItem"`
-		ManagerAccess bool  `sql:"column:bManagerAccess"`
-		OwnerAccess   bool  `sql:"column:bOwnerAccess"`
+	idsMap := make(map[int64]bool, len(itemIDs))
+	for _, itemID := range itemIDs {
+		idsMap[itemID] = true
 	}
-
-	db := s.GroupItems().MatchingUserAncestors(user).
-		Select("idItem, bManagerAccess, bOwnerAccess").
-		Where("idItem = ?", itemID).
-		Scan(&dbRes)
-	if db.Error() != nil {
-		return false, false, db.Error()
-	}
-	if len(dbRes) != 1 {
-		return false, false, nil
+	err = s.GroupItems().MatchingUserAncestors(user).
+		WithWriteLock().
+		Where("idItem IN (?) AND (bManagerAccess OR bOwnerAccess)", itemIDs).
+		PluckFirst("COUNT(DISTINCT idItem)", &count).Error()
```

Comment:
nice one 👍 
Shouldn't you write a `count(..)` fct on db to make it more readable?

Reply:
Fixed (although I'm not sure it became more readable with 'Count()')